src/state/slices/addressBookSlice/selectors.ts

```diff
@@ -0,0 +1,143 @@
+  [addressBookSlice.selectors.selectEntriesByAccountId],
+  byAccountId => Object.values(byAccountId),
+)
+
+export const selectAddressBookEntriesByChainId = createDeepEqualOutputSelector(
+  selectChainIdParamFromFilter,
+  selectAddressBookEntries,
+  (chainId, entries) => {
+    if (!chainId) return []
+
+    return entries.filter(entry => entry.chainId === chainId)
+  },
+)
+
+export const selectAddressBookEntriesBySearchQuery = createDeepEqualOutputSelector(
+  selectChainIdParamFromFilter,
+  selectSearchQueryFromFilter,
+  selectAddressBookEntriesByChainId,
+  (_, searchQuery, entries) => {
+    if (!searchQuery) return entries
+
+    const matchedEntries = matchSorter(entries, searchQuery, {
+      keys: [
+        { key: 'label', threshold: matchSorter.rankings.MATCHES },
+        { key: 'address', threshold: matchSorter.rankings.CONTAINS },
+      ],
+    })
+
+    return matchedEntries
+  },
+)
+
+/**
+ * Checks if a given address exists in the address book
+ * Handles EVM namespace matching (same address across all EVM chains)
+ */
+export const selectIsAddressInAddressBook = createDeepEqualOutputSelector(
+  selectAccountAddressParamFromFilter,
+  selectChainIdParamFromFilter,
+  selectAddressBookEntries,
+  (accountAddress, chainId, entries) => {
+    if (!accountAddress || !chainId) return false
+
+    const normalizedAddress = accountAddress.toLowerCase()
+    const { chainNamespace } = fromChainId(chainId)
+
+    // For EVM chains, check if address exists in any EVM entry
+    if (chainNamespace === CHAIN_NAMESPACE.Evm) {
+      return entries.some(entry => {
+        const entryChainNamespace = fromChainId(entry.chainId).chainNamespace
+        return (
+          entry.isExternal &&
+          entryChainNamespace === CHAIN_NAMESPACE.Evm &&
+          entry.address.toLowerCase() === normalizedAddress
+        )
+      })
+    }
+
+    // For non-EVM chains, check for exact chain + address match
+    return entries.some(
+      entry =>
+        entry.isExternal &&
+        entry.chainId === chainId &&
+        entry.address.toLowerCase() === normalizedAddress,
+    )
+  },
+)
+
+/**
+ * Checks if a given address belongs to one of the user's connected accounts
+ * Handles EVM namespace matching (same address across all EVM chains)
+ * Returns the AccountId if found, null otherwise
+ */
+export const selectInternalAccountIdByAddress = createDeepEqualOutputSelector(
+  selectAccountAddressParamFromFilter,
+  selectChainIdParamFromFilter,
+  selectAccountIdsWithoutEvms,
+  selectEvmAccountIds,
+  (accountAddress, chainId, accountIdsWithoutEvms, evmAccountIds) => {
+    if (!accountAddress || !chainId) return null
+
+    const normalizedAddress = accountAddress.toLowerCase()
+    const { chainNamespace } = fromChainId(chainId)
+
+    // For EVM chains, find first matching EVM account
+    if (chainNamespace === CHAIN_NAMESPACE.Evm) {
+      const accountId = evmAccountIds.find(accountId => {
```

Comment:
nitpick: probably want to make callback param `evmAccountId` to differentiate from upper scope, failed the gnomes jetlag litmus test and my brain couldn't differentiate `chainNamespace` and `chainId` from above scope, from this scope

Reply:
Done in [757a7dd](https://github.com/shapeshift/web/pull/10803/commits/757a7ddd6631de0719b78e63f9b94a87b7039543)